### PR TITLE
[algoim] patches to compile with clang 16

### PR DESCRIPTION
**Pull request by ericneiva**
Thank you.

### CODE REVIEW

File: A/algoim/build_tarballs.jl [hunk 1]
```diff
@@ -3,8 +3,9 @@ using BinaryBuilder
 name = "algoim"
 version = v"0.1.0"
 sources = [
-    GitSource("https://github.com/algoim/algoim.git", "979cb3b7860b53751291352a929754d525b7fde1"),
-]
+    GitSource("https://github.com/ericneiva/algoim.git", "e80020a5182597a2392194651f62fb6f4c631082"),
+] # Required to compile with Apple's clang 16. 
+# See https://github.com/algoim/algoim/pull/8 and https://github.com/algoim/algoim/pull/6
```
Comment (giordano) [inline A/algoim/build_tarballs.jl:8]: Can you please apply those as patches, rather than using a fork?  Thanks!

Comment (ericneiva) [reply A/algoim/build_tarballs.jl]: Hi, @giordano, I believe I apply them now as patches. Let me know if you meant something else. Thanks!

Comment (giordano) [reply A/algoim/build_tarballs.jl]: Yes, but the patch doesn't seem to apply.

Comment (ericneiva) [reply A/algoim/build_tarballs.jl]: Hi, @giordano, it should be ok now, one of the patches required CRLF line endings, apparently.

Let me know if I need to apply any other changes.

Thanks a lot.

File: A/algoim/bundled/patches/.gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+0002-std-result_of-deprecated-in-C-17-remove-in-C-20-std-.patch -text diff
```
Comment (giordano) [inline A/algoim/bundled/patches/.gitattributes:1]:
```suggestion
0002-std-result_of-deprecated-in-C-17-remove-in-C-20-std-.patch -text diff

```

**Pull request merged**